packages/experiments-realm/blog-category.gts

```diff
@@ -18,8 +19,8 @@ export const categoryStyle = (category: Partial<BlogCategory>) => {
     return;
   }
   return htmlSafe(`
-    background-color: ${category.backgroundColor || '#FFFFFF'};
-    color: ${category.textColor || '#000000'};
+    background-color: ${category.backgroundColor?.hexValue || '#000000'};
+    color: ${category.textColor?.hexValue || '#ffffff'};
```

Comment:
So this is the default for blog category. If I want a random default similar to how avatar computes the background color. What is the best way to do that? Wud that be an extend of this color field?

Reply:
We could make a function that converts the arbitrary string (category name) to some hex color value, in a deterministic way. And a function that returns a good contrast color to that color, similarly to how the avatar works. Then we could have a computed function that either returns the computed colors, in case the user didn't set them already